dao/src/main/java/org/thingsboard/server/dao/user/UserServiceImpl.java

```diff
@@ -201,6 +202,7 @@ public UserCredentials replaceUserCredentials(TenantId tenantId, UserCredentials
     }
 
     @Override
+    @Transactional
```

Comment:
deleteTenantAdmins and deleteCustomerUsers should be transactional as well. Otherwise the admins and customer users are deleted without transaction.